#!/usr/bin/env python
PACKAGE = "pub_sub_rate"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("number_points", int_t, 0, "Number of random points", 10,1,10000)
gen.add("motion_x", double_t, 0, "Change in Motion in X-Direction", 0.5,0,10)
gen.add("motion_y", double_t, 0, "Change in Motion in Y-Direction", 2,0,10)
gen.add("yaw", int_t, 0, "Change in Motion in Yaw [Degree]", -10,-20,20)
gen.add("number_simulator", int_t, 0, "Total time Simulator is to shown", 100,3,200)
gen.add("display_simulator", int_t, 0, "1-Show the simulator, 0- Dont show the simulator", 1,0,1)
gen.add("icp_error_threshold", double_t, 0, "Maximum ICP error Threshold for convergence", 0.0001,0,1)
gen.add("max_icp_iter", int_t, 0, "Maximum iteration for the ICP algorithm to converge", 100,1,1000)
exit(gen.generate(PACKAGE, "ros_icp", "dynamicConfig"))